packages/compass-components/src/components/context-menu.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useMemo } from 'react';
+import { Menu, MenuItem, MenuSeparator } from './leafygreen';
+import type { ContextMenuItem } from '@mongodb-js/compass-context-menu';
+import { useContextMenu } from '@mongodb-js/compass-context-menu';
+import { ContextMenuProvider as ContextMenuProviderBase } from '@mongodb-js/compass-context-menu';
+import type {
+  ContextMenuItemGroup,
+  ContextMenuWrapperProps,
+} from '@mongodb-js/compass-context-menu';
+
+export function ContextMenuProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return (
+    <ContextMenuProviderBase menuWrapper={ContextMenu}>
+      {children}
+    </ContextMenuProviderBase>
+  );
+}
+
+export function ContextMenu({ menu }: ContextMenuWrapperProps) {
+  const position = menu.position;
+  const itemGroups = menu.itemGroups;
+
+  useEffect(() => {
+    if (!menu.isOpen) {
+      menu.close();
+    }
+  }, [menu.isOpen]);
+
+  return (
+    <div
+      data-testid="context-menu"
+      style={{
+        position: 'absolute',
+        pointerEvents: 'all',
+        left: position.x,
+        top: position.y,
+      }}
+    >
+      <Menu
+        renderMode="inline"
```

Comment:
As per [the popover docs](https://github.com/mongodb/leafygreen-ui/tree/main/packages/popover#v12):

> `RenderMode.Inline` and `RenderMode.Portal` are marked deprecated and will eventually lose support. All overlay elements should migrate to using the top layer.